static/js/app.ts

```diff
@@ -1951,8 +1949,7 @@ function initializeShareProgramButtons() {
         }
 
         const response = await postJsonWithAchievements('/programs/share', {
```

Comment:
I don't see an endpoint for this request! I think the simplest is to make the loop_index and second_teachers_programs optional; by adding another route to the same function
```@app.route('/share', methods=['POST'], defaults={'loop_index': 0, 'second_teachers_programs': False })```

Reply:
What i meant is:
```suggestion
        const response = await postJsonWithAchievements(`/programs/share/${saveInfo.id}`);
```

This way you still use your existing api endpoint!